packages/core/src/internal/module-builder.ts

```diff
@@ -633,19 +638,32 @@ class IgnitionModuleBuilderImplementation<
       );
     }
 
-    if (isArtifactType(artifactOrOptions)) {
+    if (isArtifactType(addressOrArtifact)) {
+      if (
+        !(
+          typeof optionsOrAddress === "string" ||
+          isFuture(optionsOrAddress) ||
+          isModuleParameterRuntimeValue(optionsOrAddress)
+        )
+      ) {
+        this._throwErrorWithStackTrace(
+          `Invalid parameter "address" provided to contractAt "${contractName}" in module "${this._module.id}"`,
```

Comment:
This is following the pattern from `contract`.

There is a later further validation of the address field with `_contractAtFromArtifact`, but the pattern does not seem to be to pull these up, but instead to give specific type based error here.